WordPress/Classes/ViewRelated/Jetpack/Branding/Common/JetpackButton.swift

```diff
@@ -0,0 +1,128 @@
+        case .banner:
+            return .white
+        }
+    }
+
+    private func configureButton() {
+        // TODO: Remove this when the modal presentation is added
+        isUserInteractionEnabled = false
+        setTitle(Appearance.title, for: .normal)
+        tintColor = buttonTintColor
+        backgroundColor = buttonBackgroundColor
+        setTitleColor(buttonTitleColor, for: .normal)
+        titleLabel?.font = Appearance.titleFont
+        titleLabel?.adjustsFontForContentSizeCategory = true
+        titleLabel?.minimumScaleFactor = Appearance.minimumScaleFactor
+        titleLabel?.adjustsFontSizeToFitWidth = true
+        setImage(.gridicon(.plans), for: .normal)
+        contentVerticalAlignment = .fill
+        contentHorizontalAlignment = .center
+        contentMode = .scaleAspectFit
+        imageEdgeInsets = Appearance.iconInsets
+        contentEdgeInsets = Appearance.contentInsets
+        imageView?.contentMode = .scaleAspectFit
+        // banners are tipycally constrained in height, so we don't want the image to grow freely
+        if style == .badge {
+            adjustsImageSizeForAccessibilityContentSizeCategory = true
+        }
+
+        // sets the background of the jp logo to white
+        if let imageView = imageView {
+            insertSubview(imageBackgroundView, belowSubview: imageView)
+            imageBackgroundView.clipsToBounds = true
+            NSLayoutConstraint.activate([
+                imageBackgroundView.centerXAnchor.constraint(equalTo: imageView.centerXAnchor),
+                imageBackgroundView.centerYAnchor.constraint(equalTo: imageView.centerYAnchor),
+                imageBackgroundView.heightAnchor.constraint(equalTo: imageView.widthAnchor, multiplier: Appearance.imageBackgroundViewMultiplier),
+                imageBackgroundView.widthAnchor.constraint(equalTo: imageView.widthAnchor, multiplier: Appearance.imageBackgroundViewMultiplier),
+            ])
+        }
+    }
+
+    private enum Appearance {
+        static let title = NSLocalizedString("jetpack.branding.badge_banner.title", value: "Jetpack powered",
+                                             comment: "Title of the Jetpack powered badge.")
+        static let minimumScaleFactor: CGFloat = 0.6
+        static let iconInsets = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 10)
+        static let contentInsets = UIEdgeInsets(top: 6, left: 6, bottom: 6, right: 10)
+        static let maximumFontPointSize: CGFloat = 30
+        static let imageBackgroundViewMultiplier: CGFloat = 0.75
+        static var titleFont: UIFont {
+            let fontDescriptor = UIFontDescriptor.preferredFontDescriptor(withTextStyle: .body)
+            return UIFont(descriptor: fontDescriptor, size: min(fontDescriptor.pointSize, maximumFontPointSize))
+        }
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        imageBackgroundView.layer.cornerRadius = imageBackgroundView.frame.height / 2
```

Comment:
If I understand this correctly, the background view will be rounded even if the style is rectangular (`.banner`), although it'll be clear.

Could you help me understand @Gio2018 why we need to add a clear and rounded background in these cases @Gio2018? Thanks!

Reply:
Certainly. This is needed because the inside of the Jetpack logo would not have the same colors of the background (we have different combinations between dark and light), so setting this one will help achieve the logo colors we want.

Reply:
This was my mistake - it returns `.white` for `imageBackgroundColor`. 👍 Resolving this.